mavros_extras/src/plugins/play_tune.cpp

```diff
@@ -0,0 +1,59 @@
+#include <mavros/mavros_plugin.h>
+#include <mavros_msgs/PlayTuneV2.h>
+#include <cstring>
+
+namespace mavros
+{
+namespace extra_plugins
+{
+
+class PlayTunePlugin : public plugin::PluginBase
+{
+  public:
+    PlayTunePlugin() : PluginBase(), nh("~") {}
+
+    void initialize(UAS& uas_) override
+    {
+        PluginBase::initialize(uas_);
+        sub = nh.subscribe("play_tune", 1, &PlayTunePlugin::callback, this);
+    }
+
+    Subscriptions get_subscriptions() override { return { /* No subscriptions */ }; }
+
+  private:
+    ros::NodeHandle nh;
+    ros::Subscriber sub;
+
+    void callback(const mavros_msgs::PlayTuneV2::ConstPtr& tune)
+    {
+        auto msg = mavlink::common::msg::PLAY_TUNE_V2{};
+        msg.target_system = m_uas->get_tgt_system();
+        msg.target_component = m_uas->get_tgt_component();
+
+        switch (tune->format)
+        {
+            case mavros_msgs::PlayTuneV2::TUNE_FORMAT_QBASIC1_1:
+                msg.format = static_cast<uint32_t>(mavlink::common::TUNE_FORMAT::QBASIC1_1);
+                break;
+
+            case mavros_msgs::PlayTuneV2::TUNE_FORMAT_MML_MODERN:
+                msg.format = static_cast<uint32_t>(mavlink::common::TUNE_FORMAT::MML_MODERN);
+                break;
+
+            default:
+                ROS_ERROR_NAMED("play_tune", "Invalid tune format: '%i'", tune->format);
+                return;
+        }
```

Comment:
Usually that is not necessary, we just follow mavlink enums in msgs.

Reply:
Yeah, it felt a bit redundant, but at the same time, repeating the enums in the message definitions means that the message sender doesn't have to depend on mavlink, just on mavros_msgs. Is that what you meant, have the sender do `msg.format = mavlink::common::TUNE_FORMAT::QBASIC1_1` and similar?

Reply:
Looking at some other messages, I think I understand now. I'll update it to assign the format field directly from the mavros_msgs/PlayTuneV2 to the mavlink message. As long as the enum definition in the the mavros message matches the enum in mavlink, it's all good.